core/spec/models/spree/order_shipping_spec.rb

```diff
@@ -138,7 +138,7 @@ def emails
       before { Spree::OrderCancellations.new(order).short_ship([order.inventory_units.first]) }
 
       it "updates the order shipment state" do
-        expect { subject }.to change { order.reload.shipment_state }.from('ready').to('shipped')
+        expect { subject }.to change { order.reload.shipment_state }.from('pending').to('shipped')
```

Comment:
I am wrong above. I did a bit of testing with this and it doesn't try to `ship` the canceled item as [there is a filter to remove all but the on-hand ones](https://github.com/solidusio/solidus/blob/master/core/app/models/spree/order_shipping.rb#L18). I'm working right now on making another pass at the code to try to remove some of this duplicated logic while still satisfying the existing specs.